common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -222,3 +223,47 @@ def may_certify_for_course(certificates_display_behavior, certificates_show_befo
         or certificates_show_before_end
     )
     return show_early or has_ended
+
+
+def sorting_score(start, advertised_start, announcement):
+    """
+    Returns a tuple that can be used to sort the courses according
+    the how "new" they are. The "newness" score is computed using a
+    heuristic that takes into account the announcement and
+    (advertised) start dates of the course if available.
```

Comment:
I just needed to brush up on advertised vs start date.  This makes sense once you already know the rules.